### PR TITLE
add pr e2e shared build canary job

### PR DESCRIPTION
**Pull request by BenTheElder**
Adds a PR e2e using a shared bazel build so that we can make sure this works properly before switching the other jobs in the near future.

### CODE REVIEW

**Comment by BenTheElder** [comment]
/cc @krzyzacy @spxtr 
Let's see if I can not break PR testing again :upside_down_face: 

/area prow

File: prow/config.yaml [hunk 1]
```diff
@@ -112,6 +112,49 @@ presubmits:
...
+            name: ssh
+            readOnly: true
+        volumes:
+        - name: service
+          secret:
+            secretName: service-account
+        - name: ssh
+          secret:
+            defaultMode: 256
+            secretName: ssh-key-secret
```
Comment (BenTheElder) [inline prow/config.yaml]: NOTE: this spec was made from a normal prow e2e job + the bazel build args from @krzyzacy's advice, I'm not completely sure if this is correct. PTAL

File: prow/config.yaml [hunk 2]
```diff
@@ -112,6 +112,49 @@ presubmits:
...
+    - name: pull-kubernetes-e2e-gce-canary		
+      agent: kubernetes		
+      always_run: true		
+      context: pull-kubernetes-e2e-gce-canary		
+      rerun_command: "/test pull-kubernetes-e2e-gce-canary"		
+      trigger: "(?m)^/test( all| pull-kubernetes-e2e-gce-canary),?(\\s+|$)"
+      spec:
+        containers:
+        - args:
+          - --bare
```
Comment (krzyzacy) [inline prow/config.yaml]: `--bare` and `--repo` don't like each other :imp: get rid of `--bare`

Comment (BenTheElder) [reply prow/config.yaml]: done

**Comment by krzyzacy** [comment]
remove `--mode=docker` from config.json -- since it's the first prow presubmit maybe we want to guard that 'kubernetes agent' not use this flag?

**Comment by BenTheElder** [comment]
done, we probably should. I will follow up with a PR to fix up the job tests a bit.

**Comment by BenTheElder** [comment]
Test is more ore less correct, turns out there's yet another config that needed updating :-)
Hopefully once we're entirely on Prow we can consolidate the configs a bit.

File: prow/config.yaml [hunk 3]
```diff
@@ -112,6 +112,48 @@ presubmits:
...
+    - name: pull-kubernetes-e2e-gce-canary
+      agent: kubernetes
+      always_run: true
+      context: pull-kubernetes-e2e-gce-canary
+      rerun_command: "/test pull-kubernetes-e2e-gce-canary"
+      trigger: "(?m)^/test( all| pull-kubernetes-e2e-gce-canary),?(\\s+|$)"
+      spec:
+        containers:
+        - args:
+          - --timeout=75
```
Comment (krzyzacy) [inline prow/config.yaml]: tiny nit: maybe add the quote to make the args consistent

Comment (BenTheElder) [reply prow/config.yaml]: done. I was just thinking this.

**Comment by BenTheElder** [comment]
Let's see what breaks today! :upside_down_face:

**Pull request merged**